src/image_surface.rs

```diff
@@ -44,6 +110,31 @@ impl ImageSurface {
         }
     }
 
+    #[cfg(feature = "png-functions")]
+    pub fn create_from_png_stream<R: Read>(stream: R) -> io::Result<ImageSurface> {
```

Comment:
Personally I see little value in providing `Path` versions that would amount to e.g. 
```rust
create_from_png(try!(File::open(path)))
```